tests/block/block_test.go

```diff
@@ -0,0 +1,135 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2022 Authors of KubeArmor
+
+package blockposture
+
+import (
+	"fmt"
+	"time"
+
+	. "github.com/kubearmor/KubeArmor/tests/util"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+var _ = BeforeSuite(func() {
+	// install wordpress-mysql app
+	err := K8sApply([]string{"res/wordpress-mysql-deployment.yaml"})
+	Expect(err).To(BeNil())
+
+	//annotate block posture on namespace level
+	_, err = Kubectl(fmt.Sprintf("annotate ns wordpress-mysql kubearmor-network-posture=block --overwrite"))
+	Expect(err).To(BeNil())
+
+	_, err = Kubectl(fmt.Sprintf("annotate ns wordpress-mysql kubearmor-file-posture=block --overwrite"))
+	Expect(err).To(BeNil())
+
+	// delete all KSPs
+	err = DeleteAllKsp()
+	Expect(err).To(BeNil())
+
+	// enable kubearmor port forwarding
+	err = KubearmorPortForward()
+	Expect(err).To(BeNil())
+})
+
+var _ = AfterSuite(func() {
+	//remove block annotations after tests are done
+	_, err := Kubectl(fmt.Sprintf("annotate ns wordpress-mysql kubearmor-network-posture- --overwrite"))
+	Expect(err).To(BeNil())
+	_, err = Kubectl(fmt.Sprintf("annotate ns wordpress-mysql kubearmor-file-posture- --overwrite"))
+	Expect(err).To(BeNil())
+	KubearmorPortForwardStop()
+})
+
+func getWpsqlPod(name string, ant string) string {
+	pods, err := K8sGetPods(name, "wordpress-mysql", []string{ant}, 60)
+	Expect(err).To(BeNil())
+	Expect(len(pods)).To(Equal(1))
+	return pods[0]
+}
+
+var _ = Describe("Posture", func() {
+	var wp string
+	// var sql string
+
+	BeforeEach(func() {
+		wp = getWpsqlPod("wordpress-",
+			"container.apparmor.security.beta.kubernetes.io/wordpress: localhost/kubearmor-wordpress-mysql-wordpress-wordpress")
+		// sql = getWpsqlPod("mysql-",
+		// 	"container.apparmor.security.beta.kubernetes.io/mysql: localhost/kubearmor-wordpress-mysql-mysql-mysql")
+	})
+
+	AfterEach(func() {
+		KarmorLogStop()
+		err := DeleteAllKsp()
+		Expect(err).To(BeNil())
+		// wait for policy deletion
+		time.Sleep(5 * time.Second)
+	})
+
+	Describe("Policy Apply", func() {
+		It("can whitelist use of certain network protocols used by a package, such as tcp", func() {
+			// Apply policy
+			err := K8sApplyFile("res/ksp-wordpress-allow-tcp.yaml")
+			Expect(err).To(BeNil())
+
+			// Start Kubearmor Logs
+			err = KarmorLogStart("policy", "wordpress-mysql", "Network", wp)
+			Expect(err).To(BeNil())
+
+			// wait for policy creation
+			time.Sleep(5 * time.Second)
```

Comment:
sleep not required here as well as after deleting policies in `AfterEach` section. the sleep is already added to the `K8sApplyFile()`.

Reply:
okay :+1: